Open Judge System/Services/OJS.Services/Business/ProblemGroups/ProblemGroupsBusinessService.cs

```diff
@@ -40,5 +51,49 @@ public ServiceResult DeleteById(int id)
 
             return ServiceResult.Success;
         }
+
+        public ServiceResult CopyAllToContestBySourceAndDestinationContest(
+            int sourceContestId,
+            int destinationContestId)
+        {
+            if (sourceContestId == destinationContestId)
+            {
+                return new ServiceResult(Resource.Cannot_copy_problem_groups_into_same_contest);
+            }
+
+            if (!this.contestsData.ExistsById(destinationContestId))
+            {
+                return new ServiceResult(SharedResource.Contest_not_found);
+            }
+
+            if (this.contestsData.IsActiveById(destinationContestId))
+            {
+                return new ServiceResult(Resource.Cannot_copy_problem_groups_into_active_contest);
+            }
+
+            var sourceContestProblemGroups = this.contestsData
```

Comment:
Could be encapsulated in a specific data service method. Your call.

Reply:
no need to include the `AsNoTracking` in the data service, neither the `Include` statements, since they could be different for every caller and are part of the business logic